dockstore-cli-integration-testing/src/test/java/io/dockstore/client/cli/WorkflowIT.java

```diff
@@ -105,6 +113,152 @@ public void testWorkflowLaunchOrNotLaunchBasedOnCredentials() throws IOException
+        final ApiClient webClient = getWebClient(USER_2_USERNAME, testingPostgres);
+        WorkflowsApi workflowApi = new WorkflowsApi(webClient);
+
+        // Register and refresh workflow
+        Workflow workflow = workflowApi
+            .manualRegister(SourceControl.GITHUB.getFriendlyName(), "DockstoreTestUser2/md5sum-checker", "/md5sum/md5sum-workflow.cwl",
+                "test", "cwl", null);
+        Workflow refresh = workflowApi.refresh(workflow.getId());
+        Long workflowId = refresh.getId();
+        WorkflowVersion workflowVersion = refresh.getWorkflowVersions().get(0);
+        Long versionId = workflowVersion.getId();
+
+        // Download unpublished workflow version
+        workflowApi.getWorkflowZip(workflowId, versionId);
+        byte[] arbitraryURL = SwaggerUtility.getArbitraryURL("/workflows/" + workflowId + "/zip/" + versionId, new GenericType<byte[]>() {
+        }, webClient);
+        Path write = Files.write(File.createTempFile("temp", "zip").toPath(), arbitraryURL);
+        ZipFile zipFile = new ZipFile(write.toFile());
+        assertTrue("zip file seems incorrect",
+            zipFile.stream().map(ZipEntry::getName).collect(Collectors.toList()).contains("md5sum/md5sum-workflow.cwl"));
+
+        // should not be able to get zip anonymously before publication
+        boolean thrownException = false;
+        try {
+            SwaggerUtility.getArbitraryURL("/workflows/" + workflowId + "/zip/" + versionId, new GenericType<byte[]>() {
+            }, CommonTestUtilities.getWebClient(false, null, testingPostgres));
+        } catch (Exception e) {
+            thrownException = true;
+        }
+        assertTrue(thrownException);
+
+        // Download published workflow version
+        Client.main(
+            new String[] { "--config", ResourceHelpers.resourceFilePath("config_file2.txt"), "workflow", "publish", "--entry", toolpath,
+                "--script" });
+        arbitraryURL = SwaggerUtility.getArbitraryURL("/workflows/" + workflowId + "/zip/" + versionId, new GenericType<byte[]>() {
+        }, CommonTestUtilities.getWebClient(false, null, testingPostgres));
+        write = Files.write(File.createTempFile("temp", "zip").toPath(), arbitraryURL);
+        zipFile = new ZipFile(write.toFile());
+        assertTrue("zip file seems incorrect",
+            zipFile.stream().map(ZipEntry::getName).collect(Collectors.toList()).contains("md5sum/md5sum-workflow.cwl"));
+
+        // download and unzip via CLI
+        Client.main(new String[] { "--config", ResourceHelpers.resourceFilePath("config_file2.txt"), "workflow", "download", "--entry",
+            toolpath + ":" + workflowVersion.getName(), "--script" });
+        zipFile.stream().forEach((ZipEntry entry) -> {
+            if (!(entry).isDirectory()) {
+                File innerFile = new File(System.getProperty("user.dir"), entry.getName());
```

Comment:
Unless you're running in the home directory, this won't pass locally, will it? It's assuming the contents were downloaded to the home directory.

Reply:
Yes, it does look like that should be the case. It passes locally for me.